aesara/link/basic.py

```diff
@@ -1,14 +1,39 @@
-import typing
 from copy import copy, deepcopy
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    List,
+    NoReturn,
+    Optional,
+    Set,
+    Tuple,
+    Union,
+)
+
+from numpy import ndarray
 
 from aesara.configdefaults import config
-from aesara.graph.basic import Apply
+from aesara.graph.basic import Apply, Variable
 from aesara.graph.fg import FunctionGraph
 from aesara.graph.type import CType
+from aesara.graph.utils import MetaObject
 from aesara.link.utils import gc_helper, map_storage, raise_with_op, streamline
 from aesara.utils import deprecated, difference, to_return_values
 
 
+if TYPE_CHECKING:
+
+    from aesara.link.c.basic import OpWiseCLinker
+    from aesara.link.vm import VMLinker
+    from aesara.tensor.var import TensorVariable
```

Comment:
This might warrant a `# pragma: no cover`.  It'll definitely get coverage when we start running `mypy` as part of the `pre-commit` hooks, so it shouldn't be a concern.